src/engine/controls/cuecontrol.cpp

```diff
@@ -579,9 +580,12 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     // TODO(XXX) deal with spurious signals
     attachCue(pCue, pControl);
 
-    if (getConfig()->getValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false)) {
-        const QList<PredefinedColorPointer> predefinedColors = Color::kPredefinedColorsSet.allColors;
-        pCue->setColor(predefinedColors.at((hotcue % (predefinedColors.count() - 1)) + 1));
+    ConfigKey autoHotcueColorsKey("[Controls]", "auto_hotcue_colors");
+    if (getConfig()->getValue(autoHotcueColorsKey, false)) {
+        HotcueColorPaletteSettings colorPaletteSettings(getConfig());
```

Comment:
Can we cache HotcueColorPaletteSettings?
Creating a new one every time looks like cpu intensive.